src/ncp/rcp_host.cpp

```diff
@@ -438,12 +451,82 @@ const char *RcpHost::GetThreadVersion(void)
     return version;
 }
 
+static bool needRejoin(const otOperationalDatasetTlvs &aLhs, const otOperationalDatasetTlvs &aRhs)
+{
+    bool result = false;
+
+    otOperationalDataset lhsDataset;
+    otOperationalDataset rhsDataset;
+
+    SuccessOrExit(otDatasetParseTlvs(&aLhs, &lhsDataset));
+    SuccessOrExit(otDatasetParseTlvs(&aRhs, &rhsDataset));
+
+    result =
+        (lhsDataset.mChannel == rhsDataset.mChannel) &&
+        (memcmp(lhsDataset.mNetworkKey.m8, rhsDataset.mNetworkKey.m8, sizeof(lhsDataset.mNetworkKey)) == 0) &&
+        (memcmp(lhsDataset.mExtendedPanId.m8, rhsDataset.mExtendedPanId.m8, sizeof(lhsDataset.mExtendedPanId)) == 0);
+
```

Comment:
The method name is `needRejoin()`. If the channel, key, and PAN ID match, we return `true`, which indicates that we need to rejoin. Should it be the other way around? Maybe I am missing something?

Reply:
Renamed the method to `noNeedRejoin`. That's what I had meant to do.